universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -120,7 +120,7 @@ private void putNull(Path path) {
    * @param value the value to place; values of type Map will create the equivalent JSON structure
    */
   private void put(Path path, Object value) {
-    if (!hasPath(path.parentPath())) {
+    if (!path.parentPath().isEmpty() && !hasPath(path.parentPath())) {
```

Comment:
nice! suggestion (blocking): mind adding a quick test for this case, just to be safe?